java/src/main/java/com/saucelabs/simplesauce/SauceOptions.java

```diff
@@ -89,130 +213,113 @@ public String getBuild() {
+                timeoutsMap.put(Timeouts.valueOf(keyString), (Integer) val);
+            });
+            setTimeouts(timeoutsMap);
+        }
     }
 
-    public SauceOptions withIE() {
-        browserName = "IE";
-        browserVersion = "latest";
-        return this;
+    private void setEnumCapability(String key, String value) {
+        switch (key) {
+            case "browserName":
+                if (!Browser.keys().contains(value)) {
+                    String message = value + " is not a valid Browser, please choose from: " + Browser.keys();
+                    throw new InvalidSauceOptionsArgumentException(message);
+                } else {
+                    setBrowserName(Browser.valueOf(Browser.fromString(value)));
+                }
+                break;
+            case "platformName":
+                if (!SaucePlatform.keys().contains(value)) {
+                    String message = value + " is not a valid Platform, please choose from: " + SaucePlatform.keys();
+                    throw new InvalidSauceOptionsArgumentException(message);
+                } else {
+                    setPlatformName(SaucePlatform.valueOf(SaucePlatform.fromString(value)));
+                }
+                break;
+            case "jobVisibility":
+                if (!JobVisibility.keys().contains(value)) {
+                    String message = value + " is not a valid Job Visibility, please choose from: " + JobVisibility.keys();
+                    throw new InvalidSauceOptionsArgumentException(message);
+                } else {
+                    setJobVisibility(JobVisibility.valueOf(JobVisibility.fromString(value)));
+                }
+                break;
+            case "pageLoadStrategy":
+                if (!PageLoadStrategy.keys().contains(value)) {
+                    String message = value + " is not a valid Job Visibility, please choose from: " + PageLoadStrategy.keys();
+                    throw new InvalidSauceOptionsArgumentException(message);
+                } else {
+                    setPageLoadStrategy(PageLoadStrategy.valueOf(PageLoadStrategy.fromString(value)));
+                }
+                break;
+            case "unhandledPromptBehavior":
+                if (!UnhandledPromptBehavior.keys().contains(value)) {
+                    String message = value + " is not a valid Job Visibility, please choose from: " + UnhandledPromptBehavior.keys();
+                    throw new InvalidSauceOptionsArgumentException(message);
+                } else {
+                    setUnhandledPromptBehavior(UnhandledPromptBehavior.valueOf(UnhandledPromptBehavior.fromString(value)));
+                }
+                break;
+            default:
+                break;
+        }
     }
 
-    public SauceOptions withMac(MacVersion macVersion) {
-        platformName = macVersion.label;
-        browserName = "Safari";
-        return this;
+    protected String getEnvironmentVariable(String key) {
```

Comment:
Eventually, this should be private.

Reply:
The reason it is separate is to be able to mock it. I don't think we can mock it if it isn't public.

Reply:
@titusfortner agreed with @joshmgrant here. If they need to set the env variables, it's done though System Env variables rather than through hardcoding. Not sure of any good use case for allowing otherwise.